Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialTypeSourceData.cpp

```diff
@@ -252,54 +529,183 @@ namespace AZ
             return groupDefinitions;
         }
 
-        void MaterialTypeSourceData::EnumerateProperties(const EnumeratePropertiesCallback& callback) const
-        {
-            if (!callback)
+        bool MaterialTypeSourceData::BuildPropertyList(
+            const AZStd::string& materialTypeSourceFilePath,
+            MaterialTypeAssetCreator& materialTypeAssetCreator,
+            AZStd::vector<AZStd::string>& propertyNameContext,
+            const MaterialTypeSourceData::PropertyGroup* propertyGroup) const
+        {            
+            for (const AZStd::unique_ptr<PropertyDefinition>& property : propertyGroup->m_properties)
             {
-                return;
-            }
+                // Register the property...
 
-            for (const auto& propertyListPair : m_propertyLayout.m_properties)
-            {
-                const AZStd::string& groupName = propertyListPair.first;
-                const auto& propertyList = propertyListPair.second;
-                for (const auto& propertyDefinition : propertyList)
+                MaterialPropertyId propertyId{propertyNameContext, property->GetName()};
+
+                if (!propertyId.IsValid())
+                {
+                    // MaterialPropertyId reports an error message
+                    return false;
+                }
+
+                auto propertyGroupIter = AZStd::find_if(propertyGroup->GetPropertyGroups().begin(), propertyGroup->GetPropertyGroups().end(),
+                    [&property](const AZStd::unique_ptr<PropertyGroup>& existingPropertyGroup)
+                    {
+                        return existingPropertyGroup->GetName() == property->GetName();
+                    });
+
+                if (propertyGroupIter != propertyGroup->GetPropertyGroups().end())
+                {
+                    AZ_Error("Material source data", false, "Material property '%s' collides with a PropertyGroup with the same ID.", propertyId.GetCStr());
+                    return false;
+                }
```

Comment:
[question] Since this function is building a properly list, I was wondering, when it cannot find a property, or when it's colliding, if instead of returning false and make it all error if it would make sense to continue with the next property and put a warning?